### ISSUE TITLE
Output migration transforms eventEmitter.emit() into invalid output.emit()

### ISSUE THREAD

**Issue by cexbrayat** [angular/angular#58650]
### Which @angular/* package(s) are the source of the bug?

core

### Is this a regression?

Yes

### Description

The output migration transforms:

```ts
@Output() eventEmitter = new EventEmitter<string>();

// later
eventEmitter.emit();
```

into
```ts
eventEmitter = output<string>();

// later
eventEmitter.emit();
```

But the value is mandatory in the `output.emit` signature, so the application breaks.
The migration could generate `emit(undefined)` in that case, or add a `TODO` or only treat this in with the `--best-effort-mode` option.

To repro, in the following Stackblitz, run:

```
ng g @angular/core:signals --migrations=outputs --path=./ --no-best-effort-mode
```

### Please provide a link to a minimal reproduction of the bug

https://stackblitz.com/edit/stackblitz-starters-h5tea2?file=src%2Fmain.ts

### Please provide the exception or error you saw

```true
✘ [ERROR] TS2554: Expected 1 arguments, but got 0. [plugin angular-compiler]

    src/main.ts:18:22:
      18 │     this.eventEmitter.emit();
         ╵                       ~~~~

  An argument for 'value' was not provided.

    node_modules/@angular/core/index.d.ts:8288:9:
      8288 │     emit(value: T): void;
```

### Please provide the environment you discovered this bug in (run `ng version`)

```true
Angular CLI: 19.0.0-rc.1
Node: 18.20.3
Package Manager: npm 10.2.3
OS: linux x64

Angular: 19.0.0-rc.1
... animations, cli, common, compiler, compiler-cli, core, forms
... platform-browser, router

Package                         Version
---------------------------------------------------------
@angular-devkit/architect       0.1900.0-rc.1
@angular-devkit/build-angular   19.0.0-rc.1
@angular-devkit/core            19.0.0-rc.1
@angular-devkit/schematics      19.0.0-rc.1
@schematics/angular             19.0.0-rc.1
rxjs                            7.8.1
typescript                      5.5.4
zone.js                         0.14.10
```

**Comment by devversion** [comment]
I agree that we could add a TODO, or skip in "safe mode", but IMO it would also be acceptable to say that we are intentionally breaking here and surfacing a real brittle problem in the user code (caused by the bad typings of `EventEmitter` clearly).

```
@Output() eventEmitter = new EventEmitter<string>();

// later
eventEmitter.emit();
```

can already break in subtle ways at runtime, given `undefined` being used as value, but `eventEmitter.subscribe` only _receiving_ values of type _string_.

**Comment by jnizet** [comment]
I would also prefer the code to break. IMO, that's clearly a bug that should be fixed.
BTW, another godd thing to do woulmd be to document what happens when emit is called without argument: does the parent component receive an `undefined` (and thus incorrectly typed) event? Is `undefined` somehow filtered out?

**Comment by pkozlowski-opensource** [comment]
> I agree that we could add a TODO, or skip in "safe mode", but IMO it would also be acceptable to say that we are intentionally breaking here and surfacing a real brittle problem in the user code

This was my thinking while authoring this migration. I've marked this issue as a "bug" so we can insert a TODO indicating why it is breaking / what a user should do in this case.
